src/tools/storage/index.ts

```diff
@@ -117,8 +117,8 @@ export class AnyStorage {
     let response: TResponse | null | string = await backend.getItem(storageKey);
     if (response === null && !persist) {
       response = await (session
-        ? this.backends.persist.getItem(storageKey)
```

Comment:
This block handles the fallback case, if a key is not present in the choosen backend. `getBackend` will return the `persit`, `session` or `temp` backend based on `persist` and `session` arguments. So `response` will be null only if the key is not present in the selected backend. In that case we search in the next persistent level of storage which is in the order `persist` > `session` > `temp`.

Hence, if `response` is `null`, `persist` is falsy and `session` is true, the fallback storage should be `persist`.